examples/flipper/lib.rs

```diff
@@ -16,7 +16,7 @@ pub mod flipper {
 
         /// Creates a new flipper smart contract initialized to `false`.
         #[ink(constructor)]
-        pub fn default() -> Self {
+        pub fn new_default() -> Self {
```

Comment:
Why are the `new_default` changes necessary?

Reply:
New clippy lints do not allow methods with `default()` name as they can be confused with the `Default::default()` implementation. See the the previous clippy output: https://gitlab.parity.io/parity/mirrors/ink/-/jobs/2171790

Reply:
It looks like this lint's been around a [very long time](https://github.com/rust-lang/rust-clippy/blob/b3145fea6a5c9f67cf8ac77f1c9ff9bb84b71c18/clippy_lints/src/methods/mod.rs#L334), why is it only being triggered now?

Reply:
I'm not a fan of this `new_default` method, it made sense to have a `default` method for the contract. Maybe we can work around it using a the `Default` trait impl instead?

Reply:
> I'm not a fan of this `new_default` method, it made sense to have a `default` method for the contract. Maybe we can work around it using a the `Default` trait impl instead?

Can you open a follow up issue for this please?